spock-core/src/main/java/org/spockframework/runtime/RunContext.java

```diff
@@ -229,4 +229,24 @@ static RunContext createBottomContext() {
     List<Class<?>> configs = extensionClassesLoader.loadConfigClassesFromDefaultLocation();
     return new RunContext("default", spockUserHome, script, classes, configs);
   }
+
+  public void ensureInstalled() {
+    LinkedList<RunContext> runContexts = contextStacks.get();
+    if (runContexts.isEmpty()) {
+      // we got a new thread that didn't inherit the original context, so just add us here
+      runContexts.add(this);
+    } else {
+      if (!runContexts.contains(this)) {
+        // something weird happened, and we got the got a separate RunContext hierarchy from what we'd expect
+        // maybe some thread re-use between runs
+        runContexts.clear();
+        runContexts.add(this);
+      } else {
```

Comment:
This seams a bit too much to always asks for contains, which will traverse the whole linked list, and after that we traverse again. Isn't there a better way to do that?

Or maybe this is premature opt from my side, because are there many RunContexts anyway?

Reply:
I think in the wild it is 99.9% just one context.
And for Spock it is 99.0% 1 and 1% with 2. 
And it should be the first element if nothing has gone wrong.